### PR TITLE
[ADF-4577] Should display the validation message when Process definition has name property set to null

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/process-services-cloud/start-process-cloud.e2e.ts [hunk 1]
```diff
@@ -92,4 +93,17 @@ describe('Start Process', () => {
         processCloudDemoPage.processListCloudComponent().checkContentIsDisplayedByName(processName);
 
     });
+
+    it('[C309875] Should display the validation message when Process definition has name property set to null', () => {
+        appListCloudComponent.checkAppIsDisplayed(simpleApp);
+        appListCloudComponent.goToApp(simpleApp);
+        processCloudDemoPage.openNewProcessForm();
```
Comment (marouanbentaleb) [inline e2e/process-services-cloud/start-process-cloud.e2e.ts]: Can we have those 3 lines in a beforeEach ?

File: lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts [hunk 2]
```diff
@@ -114,4 +114,12 @@ export class StartProcessCloudPage {
             }
         });
     }
+
+    selectOptionAtIndex(index) {
```
Comment (marouanbentaleb) [inline lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts]: I think we can have a more significant name here

Comment (DenysVuika) [reply lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts]: type information is missing
```suggestion
    selectOptionAtIndex(index: number) {
```

Comment (gmandakini) [reply lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts]: can't use a name as we are referring ot a process definition with no name.

File: e2e/process-services-cloud/start-process-cloud.e2e.ts (hunk 1 above)
Comment (gmandakini) [reply e2e/process-services-cloud/start-process-cloud.e2e.ts]: we can't in this scenario, as we are doing things in afterEach to come to the main apps page and this way the test is  consistent in run results.

File: lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts [hunk 3]
```diff
@@ -114,4 +114,12 @@ export class StartProcessCloudPage {
             }
         });
     }
+
+    selectProcessDefinitionAtIndex(index: number) {
+        this.processDefinition.click();
```
Comment (eromano) [inline lib/testing/src/lib/process-services-cloud/pages/start-process-cloud-component.page.ts]: can you use the common browser action in the  /lib/core/utils/browser-actions.ts to sendKeys?

**Comment by gmandakini** [comment]
> see my comment above , plus why are you uploading a new simpleapp ?

I needed to add few more process definitions to simpleapp for some of the pending Jira tickets automation + this one as well. Hence adding a new app definition.

**Comment by gmandakini** [comment]
Closed this as there were too many conflicts and opened a new pr => https://github.com/Alfresco/alfresco-ng2-components/pull/4802/files

**Pull request closed without merging**